components/linking_landing_page/linking_landing_page.tsx

```diff
@@ -0,0 +1,473 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import {FormattedMessage} from 'react-intl';
+import safeOpenProtocol from 'custom-protocol-detection';
+
+import desktopImg from 'images/deep-linking/deeplinking-desktop-img.png';
+import mobileImg from 'images/deep-linking/deeplinking-mobile-img.png';
+import MattermostLogoSvg from 'images/logo.svg';
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import CheckboxCheckedIcon from 'components/widgets/icons/checkbox_checked_icon';
+import BrowserStore from 'stores/browser_store';
+import {LandingPreferenceTypes} from 'utils/constants';
+import * as Utils from 'utils/utils';
+
+import * as UserAgent from 'utils/user_agent';
+
+type Props = {
+    defaultTheme: any;
+    desktopAppLink?: string;
+    iosAppLink?: string;
+    androidAppLink?: string;
+    siteUrl?: string;
+    siteName?: string;
+    brandImageUrl?: string;
+    enableCustomBrand: boolean;
+}
+
+type State = {
+    protocolUnsupported: boolean;
+    browserUnsupported: boolean;
+    rememberChecked: boolean;
+    redirectPage: boolean;
+    location: string;
+    nativeLocation: string;
+    brandImageError: boolean;
+    navigating: boolean;
+}
+
+export default class LinkingLandingPage extends PureComponent<Props, State> {
+    constructor(props: Props) {
+        super(props);
+
+        const location = window.location.href.replace('/landing#', '');
+
+        this.state = {
+            protocolUnsupported: false,
+            browserUnsupported: false,
+            rememberChecked: false,
+            redirectPage: false,
+            location,
+            nativeLocation: location.replace(/^(https|http)/, 'mattermost'),
+            brandImageError: false,
+            navigating: false,
+        };
+    }
+
+    componentDidMount() {
+        Utils.applyTheme(this.props.defaultTheme);
+        if (this.checkLandingPreferenceApp()) {
+            this.openMattermostApp();
+        }
+
+        window.addEventListener('beforeunload', this.clearLandingPreferenceIfNotChecked);
+    }
+
+    clearLandingPreferenceIfNotChecked = () => {
```

Comment:
Instead of clearing this preference when we navigate, why don't we just only save the value before navigation? It seems a little weird to save the value back to local storage and the clear it if the user doesn't actually click the button to navigate

Reply:
We also save the value when we open the desktop app, which doesn't navigate. This gives us full consistent control over when the value is saved, even if it feels a little awkward.